lib/fluent/plugin/out_s3.rb

```diff
@@ -80,6 +82,15 @@ def start
       else
         options[:credential_provider] = AWS::Core::CredentialProviders::EC2Provider.new({:retries => @aws_iam_retries})
       end
+
+      # If a role ARN was provided, use the EC2 credential provider to connect to STS.
+      # Then use an AssumeRole provider when connecting to S3.
+      if @sts_role_arn
+        sts_conn = AWS::STS.new(options)
```

Comment:
Shouldn't we also use other configurable parameters `proxy_uri`, `use_ssl`, etc here?